parlai/chat_service/services/websocket/agents.py

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import time
+import logging
+from queue import Queue
+
+from parlai.core.agents import Agent
+
+
+class WebsocketAgent(Agent):
+    """Class for a person that can act in a ParlAI world via websockets"""
+
+    def __init__(self, opt, manager, task_id, socketID):
+        super().__init__(opt)
+        self.manager = manager
+        self.id = socketID
+
+        self.active = True
+        self.msg_queue = Queue()
+        self.stored_data = {}
+        self.task_id = task_id
+        self.message_request_time = None
+        self.set_stored_data()
+
+    def set_stored_data(self):
+        """Gets agent state data from manager"""
+        agent_state = self.manager.get_agent_state(self.id)
+        if agent_state is not None and hasattr(agent_state, 'stored_data'):
+            self.stored_data = agent_state.stored_data
+
+    def observe(self, act):
+        """Send an agent a message through the websocket manager"""
+        logging.info(f"Sending new message: {act}")
+        if 'payload' in act:
+            raise ValueError("Payload not supported yet by websockets")
+        else:
+            msg = act['text']
+            self.manager.observe_message(self.id, msg)
+
+    def put_data(self, message):
+        """Put data into the message queue if it hasn't already been seen"""
```

Comment:
should we be checking that the message has not already been seen?

Reply:
Updated the docs to reflect that we don't do the check. 

`get_new_act_message` removes the message from the queue and the `on_new_message` callback for WebSockets is only be called once per message, so duplicate messages shouldn't be an issue.